### PR TITLE
Fix/high dpi display

### PR DESCRIPTION
**Pull request by KIC**
resolve dpi conflicts

### CODE REVIEW

File: libqtile/pangocffi.py [hunk 1]
```diff
@@ -185,3 +187,10 @@ def parse_markup(value, accel_marker=0):
 def markup_escape_text(text):
     ret = gobject.g_markup_escape_text(text.encode('utf-8'), -1)
     return ffi.string(ret).decode()
+
+
+def set_default_dpi(dpi):
+    # Note: this is per-thread. However, we're single threaded, so it's fine.
```
[the remark this answers is not in the document]
Comment (KIC) [reply libqtile/pangocffi.py]: Hmm ok. I have tried to pass along the dpi via the drawer object. Sadly the result is the same. Is it really such a speciality to have high dpi displays these days?

[the remark this answers is not in the document]
Comment (KIC) [reply libqtile/pangocffi.py]: i have tried it in the manager as well:
    def _configure(self):
        """
        This is the part of init that needs to happen after the event loop is
        fully set up. asyncio is required to listen and respond to backend
        events.
        """
        self._process_screens()
        self.current_screen = self.screens[0]

        self.conn.flush()
        self.conn.xsync()
        self.core._xpoll()
        pangocffi.set_default_dpi(self.dpi)
        
still looks unchanged

[the remark this answers is not in the document]
Comment (KIC) [reply libqtile/pangocffi.py]: @tych0 can you eventually tell me all the places where I would have to hard code 150 dpi. I will do the rest with configuration and testing.

[the remark this answers is not in the document]
Comment (tych0) [reply libqtile/pangocffi.py]: Unfortunately I don't know them all :). But I don't think we want to hard code anything anyway. Querying the DPI isn't that hard, it's mostly about fixing up the math after custom DPI settings, which we may as well do programmatically vs. hard coding different DPI values.

[the remark this answers is not in the document]
Comment (KIC) [reply libqtile/pangocffi.py]: yeah I get that, but I want to bring in the nice configurable and tested solution once I get it working with hard coded values. such that I can understand where in the code dpi's play a role. currently I don't. when I use xypher server I was not able to change any rendering behaviour so far.

[the remark this answers is not in the document]
Comment (KIC) [reply libqtile/pangocffi.py]: interestingly not even the unchanged old dpi branch works as i would expect. the images look exactly the same while i would expect a bigger bar with a higher dpi setting rendered on the same display.

dpi_scale_1
dpi_scale_2

[the remark this answers is not in the document]
Comment (tych0) [reply libqtile/pangocffi.py]: It's not clear the bar itself should scale with DPI -- users hard code the size of their bars now depending on personal preference and other things. Maybe we can ignore that in the face of different DPIs, but probably we shouldn't. So I think that's ok, but the math about what things were clicked is definitely wrong.

[the remark this answers is not in the document]
Comment (KIC) [reply libqtile/pangocffi.py]: this is still an issue. I think everything rendered should scale appropriately scaled. the bar as well. sure you can set the font size and workaround, that's what I am doing as well. however not every app allows you to adjust the font size. like teams (yes I know but my company decided to use it). so this really is an issue.

**Comment by KIC** [comment]
I don't think this pr is worth anything, I can't get it to work 😭

**Pull request closed without merging**